docs/details/signal.dox

```diff
@@ -211,32 +211,108 @@ respectively, then the possible batch operations are as follows.
 \defgroup signal_func_approx1 approx1
 \ingroup approx_mat
 
-approx1 interpolates data along the first dimensions.
-It has three options for the type of interpolation to perform:
-- Nearest neighbor  - \ref AF_INTERP_NEAREST
-- Linear interpolation  - \ref AF_INTERP_LINEAR
-- Bilinear interpolation - \ref AF_INTERP_BILINEAR
-- Cubic interpolation - \ref AF_INTERP_CUBIC
-
-Interpolation is performed assuming input data is equally spaced with indices
-in the range [0, n). The positions are sampled with respect to data at these
-locations.
+approx1 interpolates data along one dimension. Interpolation computes
+for values of unknown points out of known ones.
+
+The multi-dimensional (>=1D) input array, `in`, contains known values
+to be used for interpolation. These known values lie on a 1D
+interpolation grid - a one-dimensional, uniformly-spaced grid of
+points. Interpolation is performed to obtain values for any unknown
+points on the grid. The index values of the grid can be specified with
+the `grid_beg` and `grid_step` arguments.
+
+The position array (<=1D), `pos`, contains a combination of known and
+unkown positions/indices on the interpolation grid. The position
+array, `pos`, must be one-dimensional and must lead across the
+dimension in which the measurements will be made in the input array.
+
+Any unknown position/index values that lie within the interpolation
+grid range will be calculated for using interpolation. Unknown points
```

Comment:
What if we replace "calculated for" with "determined" (and "computes for" with "determines" in the very first line)?